src/libraries/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableArray.cs

```diff
@@ -181,11 +181,7 @@ public static ImmutableArray<T> Create<T>(T[] items, int start, int length)
             }
 
             var array = new T[length];
-            for (int i = 0; i < array.Length; i++)
-            {
-                array[i] = items[start + i];
-            }
-
+            Array.Copy(items, start, array, 0, length);
```

Comment:
What is the break-even point for this? This will regress small sizes.

Reply:
I'll setup a benchmark for this.

Reply:
Answered in a comment below

Reply:
So it makes some cases slower and some faster. I think we would need to additional data to tell whether this is an improvement on average. E.g. we would need data about how this method is used by Roslyn that is a heavy user of ImmutableArrays to make sure that this won't regress Roslyn performance.

The rest of the PR looks good to me. Could you please revert this change so that we can take the rest?

Reply:
I can definitely split these across multiple atomic PRs so we tackle each different problem individually